tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,208 @@
+import math
+import pytest
+from decimal import Decimal
+import datetime
+from datetime import timezone, timedelta
+import trino
+
+
+@pytest.fixture
+def trino_connection(run_trino):
+    _, host, port = run_trino
+
+    yield trino.dbapi.Connection(
+        host=host, port=port, user="test", source="test", max_attempts=1
+    )
+
+
+def test_int_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="false", python=False) \
+        .add_field(sql="cast(null AS BOOLEAN)", python=None) \
+        .add_field(sql="cast(127 AS TINYINT)", python=127) \
+        .add_field(sql="cast(-128 AS TINYINT)", python=-128) \
```

Comment:
Should also test some value not at the edge of supported range.

Reply:
Done

Reply:
You mentioned at one point is that the order of the tests should be null / min value / intermediate value / max value. So should we be testing the min value or not?

Reply:
We should test min value. What the later comment was referencing is https://github.com/trinodb/trino-python-client/pull/206/commits/eba160bdf600d96d12d72c41cd77fdb9c6fe52fb#diff-da781ffd6f3bd2fa7b1f7d7055dac6ce9f04b9a776feb777e35247eb024719a7L30-L31 (which you've fixed now).